packages/one-app-dev-bundler/esbuild/plugins/externals-loader.js

```diff
@@ -40,20 +74,74 @@ const externalsLoader = ({ bundleType }) => ({
     // return a namespace. (see this above)
     // your onLoad can then just match .* within that namespace and you guarantee you target
     // every package you want.
-    build.onLoad({ filter: /.*/, namespace: 'externalsLoader' }, async (args) => {
-      const jsContent = `\
-try {
-  module.exports = ${globalReferenceString}.getTenantRootModule().appConfig.providedExternals['${args.path}'].module;
-} catch (error) {
-  const errorGettingExternal = new Error('Failed to get external ${args.path} from root module');
-  errorGettingExternal.shouldBlockModuleReload = false;
-  throw errorGettingExternal;
-}
-`;
+    build.onLoad({ filter: /.*/, namespace: 'externalsLoader' }, async ({ path: externalName, ...args }) => {
+      if (bundleType === BUNDLE_TYPES.SERVER) {
+        const require = createRequire(import.meta.url);
+        const cjsCode = (await swc.transform(await fs.promises.readFile(require.resolve(externalName), 'utf8'), {
+          jsc: {
+            target: 'es5',
+          },
+        })).code;
+
+        const resolveDir = ignoreUntil(
+          (value) => value === 'node_modules',
+          require
+            .resolve(externalName)
+            .split(path.sep)
+            .reverse()
+        )
+          .reverse()
+          .join(path.sep);
+
+        return {
```

Comment:
This is a pretty big if statement to then do an early return. Its unclear that the execution would not pickup at line 118.

Can you abstract the two large code paths into functions?

```js
if (bundleType === BUNDLE_TYPES.SERVER) {
  return serverExternalsLoader(...);
} else {
  return browserExternalsLoader(...);
}
```

Reply:
agree, I will leave it as is right now since I'm refactoring the node server external fallback part, and that refactoring will simplify the implementation